### PR TITLE
Keep thread names under 15 characters

### PR DESCRIPTION
**Pull request by ob-stripe**
### Description
Keep thread names under 15 characters.

Linux thread names are limited to 15 characters. Trying to set a longer name will just silently do nothing 😡 

This PR ensures that all threads created by Puma have names that are 15 characters or shorter. Who needs vowels anyway.

### Your checklist for this pull request
- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [x] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: lib/puma/runner.rb [hunk 1]
```diff
@@ -69,7 +69,7 @@ def start_control
 
       control.binder.parse [str], self, 'Starting control server'
 
-      control.run thread_name: 'control'
+      control.run thread_name: 'ctl'
```
Comment (nateberkopec) [inline lib/puma/runner.rb:72]: Suggestion: lets not change any thread names that are always less than 15 characters.

Comment (ob-stripe) [reply lib/puma/runner.rb:72]: The problem is that this thread name is also used as a prefix for its accompanying threadpool threads. So if we don't shorten it, we end up with names like `puma control tp 001` which is longer than 15 characters.

**Comment by ob-stripe** [comment]
I added a Linux-only test that uses `ps` to assert that the thread name is correctly set at the OS level.

**Comment by dentarg** [comment]
Looks like the added test doesn't work on jruby/truffleruby (but do in their -head versions? 🤔) Even fails when running only that test `bundle e m test/test_thread_pool.rb:65` (testing in `docker run --rm -it -v $PWD:/app -w /app ghcr.io/flavorjones/truffleruby:21.3.0 bash`), any ideas @ob-stripe?

**Comment by dentarg** [comment]
Ah I guess we can't see the thread names in `/proc` when we run on the JVM

**Comment by dentarg** [comment]
Okay, always one step behind @MSP-Greg (https://github.com/puma/puma/commit/fa178f3c829610d82b9c5013db748ff0207e5773) 😄

**Comment by ob-stripe** [comment]
Thanks! I was struggling to get the test suite to pass with all different environments.

**Pull request merged**